packages/task/src/browser/task-frontend-contribution.ts

```diff
@@ -83,13 +83,13 @@ export class TaskFrontendContribution implements CommandContribution, MenuContri
                 isEnabled: () => true,
                 // tslint:disable-next-line:no-any
                 execute: (...args: any[]) => {
-                    let type: string | undefined;
+                    let source: string | undefined;
                     let label: string | undefined;
                     if (args) {
```

Comment:
I think this if check should be updated, now that `args` will always be an array, and that `[]` evaluates to `true`.

Seeing how you fixed the issue -where `source` and `label` can be undefined because of the way things currently are-, you can replace this handler by:

```ts
            {
                isEnabled: () => true,
                // tslint:disable-next-line:no-any
                execute: (...args: any[]) => {
                    const [source, label] = args;
                    if (source && label) {
                        return this.taskService.run(source, label);
                    }
                    return this.quickOpenTask.open();
                }
            }
```